engine/runtime-instrument-id-execution/src/main/java/org/enso/interpreter/instrument/IdExecutionInstrument.java

```diff
@@ -302,12 +305,17 @@ public EventBinding<ExecutionEventListener> bind(
       Consumer<IdExecutionInstrument.ExpressionValue> onComputedCallback,
       Consumer<IdExecutionInstrument.ExpressionValue> onCachedCallback,
       Consumer<Exception> onExceptionalCallback) {
-    SourceSectionFilter filter =
-        SourceSectionFilter.newBuilder()
-            .tagIs(StandardTags.ExpressionTag.class, StandardTags.CallTag.class)
-            .tagIs(IdentifiedTag.class)
-            .sourceSectionEquals(locationFilter.getSections())
-            .build();
+    var builder = SourceSectionFilter.newBuilder()
+          .tagIs(StandardTags.ExpressionTag.class, StandardTags.CallTag.class)
+          .tagIs(IdentifiedTag.class)
+          .tagIsNot(AvoidIdInstrumentationTag.class)
+          .sourceIs(module::isModuleSource);
+
+    if (entryCallTarget instanceof RootCallTarget r && r.getRootNode() instanceof ClosureRootNode c && c.getSourceSection() != null) {
+      SourceSection s = c.getSourceSection();
+      builder.lineIn(SourceSectionFilter.IndexRange.between(s.getStartLine(), s.getEndLine() + 1));
+    }
```

Comment:
Still, it's not clear why we're falling back to location filter instead of using `AvoidInstrumentation` tag. We usually document all non-trivial logic in the codebase, and this one also requires a comment.

Reply:
Is the meaning better with fd6702509?